osu.Game/Database/RealmObjectExtensions.cs

```diff
@@ -102,13 +91,28 @@ public static class RealmObjectExtensions
                      }
                  }
              });
-            c.CreateMap<BeatmapSetInfo, BeatmapSetInfo>()
-             .MaxDepth(2)
-             .AfterMap((s, d) =>
-             {
-                 foreach (var beatmap in d.Beatmaps)
-                     beatmap.BeatmapSet = d;
-             });
+        }).CreateMapper();
+
+        /// <summary>
+        /// A slightly optimised mapper that avoids double-fetches in cyclic reference.
+        /// </summary>
+        private static readonly IMapper beatmap_set_mapper = new MapperConfiguration(c =>
+        {
+            applyCommonConfiguration(c);
+
+            c.CreateMap<BeatmapInfo, BeatmapInfo>()
+             .MaxDepth(1)
+             .ForMember(b => b.BeatmapSet, cc => cc.Ignore());
```

Comment:
Just to confirm my understanding - while `BeatmapSet` is ignored here, it will still be populated by the `AfterMap()` config for `BeatmapSetInfo` coming from `applyCommonConfiguration()`, right?

Reply:
Sounds correct. Since this is the mapper configuration for detaching `BeatmapSetInfo` specifically, child `BeatmapInfo`s would ignore mapping `BeatmapSet`, and on the `AfterMap` it gets assigned accordingly.

Reply:
I've added an inline comment explaining this.